### PR TITLE
Xaiki/provider opts

### PR DESCRIPTION
**Pull request by xaiki**
provider ops rebased, it seems to work.
makes lang_dropdown do what it should.
i'm seeing very slow metadata times, but it shouldn't be related to what this touches...

### CODE REVIEW

File: src/app/global.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ var
     inherits = require('util').inherits,
     Q = require('q'),
 
+    // es6 pollyfills
+    objectAssign = require('es6-object-assign').assign,
```
Comment (vankasteelj) [inline src/app/global.js]: Since we moved to nw18, we don't need a polyfill I think.

File: src/app/lib/models/generic_collection.js [hunk 2]
```diff
@@ -52,7 +52,8 @@
 
                         return;
                     }
-                    movie.provider = torrentProvider.name;
+
+                    movie.provider = torrentProvider;
```
Comment (vankasteelj) [inline src/app/lib/models/generic_collection.js]: Should remain torrentProvider.name, torrentProvider is an object. Or if you want to change that (see comment below), there is more than just here

File: src/app/lib/views/lang_dropdown.js [hunk 3]
```diff
@@ -1,47 +1,80 @@
 (function (App){
     'use strict';
 
+    var objectAssign = require('es6-object-assign').assign;
```
Comment (vankasteelj) [inline src/app/lib/views/lang_dropdown.js]: There is no need to defined twice objectAssign (if defined in global, it's already accessible)

File: src/app/lib/views/browser/item.js [hunk 4]
```diff
@@ -164,7 +158,7 @@
             var realtype = this.model.get('type');
             var itemtype = realtype.replace('bookmarked', '');
             var modelType = itemtype.charAt(0).toUpperCase() + itemtype.slice(1); // 'Movie', 'Show'
-            var provider = App.Providers.get(this.model.get('provider'));
+            var provider = this.model.get('provider');
```
Comment (vankasteelj) [inline src/app/lib/views/browser/item.js]: Related to the above comment: if the backbone model has to support the entire provider functions (before there was just a string attached), isnt that memory-consuming? I mean, the provider is quite an heavy object to pass attached to a model, isnt it? Anyway, if this is really wanted, then I think there are other places where model.attributes.provider is still using a string (all should be ported, for consistency)

File: src/app/templates/play-control.tpl [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<div class="play-control">
```
Comment (vankasteelj) [inline src/app/templates/play-control.tpl]: ok, this is split in "component", great, but there is no css attached to it. Shouldnt there be a play-control.styl?

**Comment by vankasteelj** [comment]
Remarks (while using)

1) Visually: at 960x580 (min WxH for the window), it's ugly: 
nw_2016-12-04_11-33-50

2) backdrop is broken on moviedetails (stays black)

3) provider.detail is not a function error, probably linked to your changes to provider above.
- enter details
- close details
- you can't enter a 2nd time

File: src/app/lib/views/play_control.js [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        switchAudio: function (lang) {
+            var audios = this.model.get('audios');
+
+            if (audios === undefined || audios[lang] === undefined) {
+                lang = 'none';
+            }
+
+            this.audio_selected = lang;
+
+            console.info('Audios: ' + this.audio_selected);
```
Comment (vankasteelj) [inline src/app/lib/views/play_control.js]: this is called twice on every load, coulnt figure out why

File: src/app/global.js [hunk 7]
```diff
@@ -5,9 +5,6 @@ var
     inherits = require('util').inherits,
     Q = require('q'),
 
-    // es6 pollyfills
-    objectAssign = require('es6-object-assign').assign,
```
Comment (xaiki) [inline src/app/global.js]: cool, did we include that in package.json?

Comment (vankasteelj) [inline src/app/global.js]: There is no need, Object.assign is there in nwjs >= 13, through updated node

**Pull request merged**